### ISSUE TITLE
Can we make search modify the URL

### ISSUE THREAD

**Issue by EssamWisam** [JuliaDocs/Documenter.jl#2543]
I was willing to integrate documenter's search with an external site for one package that I'm working on. 
<img width="1373" alt="image" src="https://github.com/JuliaDocs/Documenter.jl/assets/49572294/7546a07e-3bc4-4b36-9cf6-3be40eff49d6">
However, that seems impossible because the current search implementation doesn't the URL in any way. It would be nice to have search function similar to common search engines, where the url is modified based on the input.

**Comment by mortenpi** [comment]
We actually used to do that, and I think it should doable with the new search as well, where we just look at and update a special query parameter.

> integrate documenter's search with an external site

Do you mean you want to be able to link to the search results? Or do you mean something else?

**Comment by EssamWisam** [comment]
> We actually used to do that, and I think it should doable with the new search as well, where we just look at and update a special query parameter.
> 
> > integrate documenter's search with an external site
> 
> Do you mean you want to be able to link to the search results? Or do you mean something else?

Yes, to be able to link it with the link being meaningful. An example from Google:
```
https://www.google.com/search?q=julia+documenter
```
First, it can be linked and second, it allows integration with another site since any external site can do 
```
https://www.google.com/search?q=${custom-search-term}
```

**Comment by mortenpi** [comment]
Pre-1.0 that used to be possible: https://documenter.juliadocs.org/v0.27/search/?q=deploydocs

I'd say we can bring this back if someone wants to take a stab at implementing it. The only caveat is that there wouldn't be  a dedicated search page, but it would probably work on every page. I guess the canonical choice can just be the root page though.

**Comment by Hetarth02** [comment]
Happy to take a try at this.

**Comment by Rahban1** [comment]
would love to work on this.
